### PR TITLE
Codec from Fleece

### PR DESCRIPTION
**Pull request by wallymathieu**
Things we can note: The code becomes more safe compared to creating `mapFromHashEntries` and `mapToHashEntries` separately. The same can be seen in Fleece.

It would be nice to be able to describe a codec without having to have a "create" function. But to begin with this form has enough benefits to make sense.

### CODE REVIEW

File: Auctions/MapToRedis.fs [hunk 1]
```diff
@@ -3,146 +3,121 @@ open Auctions.Domain
...
+
+  let diApply combiner toBC (remainderFields: SplitCodec<'S, 'f ->'r, 'T>) (currentField: Codec<'S, 'f>) = //NOTE: From Fleece
+      (
+          Compose.run (Compose (fst remainderFields: Decoder<'S, 'f -> 'r>) <*> Compose (fst currentField)),
+          toBC >> (encode (snd currentField) *** encode (snd remainderFields)) >> combiner
+      )
+
+  /// <summary>Appends a field mapping to the codec.</summary>
+  /// <param name="fieldName">A string that will be used as key to the field.</param>
+  /// <param name="getter">The field getter function.</param>
```
Comment (gusty) [inline Auctions/MapToRedis.fs]: `fromRedis` and `toRedis` params are missing in this xmldocs

File: Auctions/AppendAndReadBatchRedis.fs [hunk 2]
```diff
@@ -12,24 +12,24 @@ type AppendAndReadBatchRedis(connStr:string) =
   let hashCreate (batch : IBatch) command = 
     let id = Guid.NewGuid().ToString("N")
     let entries = mapToHashEntries command
-    batch.HashSetAsync(redisKey id, entries |> List.toArray) |> ignore
+    batch.HashSetAsync(RedisKey.op_Implicit id, entries |> List.toArray) |> ignore
```
Comment (gusty) [inline Auctions/AppendAndReadBatchRedis.fs]: Maybe you can use `implicit` from F#+ have you tried it?

Comment (wallymathieu) [reply Auctions/AppendAndReadBatchRedis.fs]: 👍

File: Auctions/MapToRedis.fs (hunk 1 above)
Comment (wallymathieu) [reply Auctions/MapToRedis.fs]: I've added them 👍

**Pull request merged**